microsoft/botframework-solutions
# [Skill Template][TypeScript] Add Skill Template

## Description
- Remove **skill generator** from `Skill-Template` folder
- Add `Skill Template`
  - Add **main functionality** for the generator skill
  - Implement **TSLint**
  - Add basic configuration for the template
  - Add **deployment resources**
  - Add **dialogs** and **responses**
  - Add **services** folder
  - Add **startup files** and **adapter**
  - Add `npm registry`
  - Update related documentations
  - Update the logic of `copier.js` to make a dynamic `solutionName` property
  - Update `myget` references to https://botbuilder.myget.org/F/aitemplates/npm/
  - Update `botbuilder` libraries to version `4.3.4`
- Add **main functionality** for the generator assistant


Related to #1199 

## Testing Steps

1. Go to `AI\templates\Virtual-Assistant-Template\src\typescript\generator-botbuilder-assistant`
2. Open a terminal in this location
3. Run `npm link` to symlink the package folder
4. Run`yo botbuilder-assistant:skill`
5. Check that the generator finished successfully
6. Run the `deploy.ps1` (which is inside `deployment\scripts` folder) being in the generated skill folder
7. Verify that the deployment finished successfully
8. Run your generated skill
9. Check that the customSkill has the same behavior as C# version

## Checklist
